### ISSUE TITLE
Error message output when trying to autocomplete

### ISSUE THREAD

**Issue by peey** [fish-shell/fish-shell#7176]
fish version: 3.1.0
OS: Ubuntu 20.04
$TERM: screen (actually I'm using tmux, inside default terminal app)

Sometimes, when I type in something that fish can't autocomplete but I hit tab anyways, I see this garbage output : `__fish_is_first_arg: command not found
in command substitution`

![image](https://user-images.githubusercontent.com/4688870/86508743-1207cd00-be00-11ea-9b07-8b221a291a2b.png)

Fish should just do nothing in such cases, so I can complete typing the command on my own

**Comment by faho** [comment]
This happens because fish can't find the function, which means it is either misspelled (which is a bug in the completion script) or your install is broken or misconfigured.

Please give the output of `type __fish_is_first_arg`, `echo $fish_function_path` and `cat $fish_function_path/__fish_is_first_arg.fish`.

(Also note that Ubuntu's package is out of date, fish 3.1.2 is out with important fixes - we offer [a PPA](https://launchpad.net/~fish-shell/+archive/ubuntu/release-3))

**Comment by peey** [comment]
![image](https://user-images.githubusercontent.com/4688870/86526576-6ae46d80-beb3-11ea-949e-9344d0b7929c.png)

You are correct. I'm not sure what broke it. Could it be that I configured my install with dotfiles from an older version of fish? 

Could you please tell me how I should fix this?

**Comment by faho** [comment]
Okay, your fish installation is broken. `__fish_is_first_arg` should be in the fish *package*, which you need to reinstall. Something like `apt install --reinstall fish`.

Also check your drives for any possible failures - that might have caused the file to disappear.

**Comment by peey** [comment]
I figured that it should appear in `/usr/share/fish/functions`. Re-installing from apt didn't work, but installing from the PPA did work. Thanks!

I'm still not sure what caused the issue. To investigate, I tested it on docker instance of `ubuntu:20.04`. I'm not sure if these results reflect state of the usual ubuntu installs on computers.

I just followed these steps:

```
docker run --rm -it ubuntu:20.04
apt-get apt-get install -y fish
fish
type __fish_is_first_arg
```

And got

![image](https://user-images.githubusercontent.com/4688870/86526897-32469300-beb7-11ea-9f8f-d4511e046781.png)

**Comment by mqudsi** [comment]
Re-opened until the state of the downstream packages can be asserted one way or the other.

**Comment by zanchey** [comment]
It's definitely wrong downstream, the Debian packages from 3.1.0+ don't include that file for some reason ([Ubuntu](https://packages.ubuntu.com/focal/all/fish-common/filelist), [Debian](https://packages.debian.org/bullseye/all/fish-common/filelist)).

**Comment by zanchey** [comment]
Oh, hahah. It's because the Debian packages pass `-Xrg.fish` to `dh_install` to exclude the `rg.fish` file, but that's actually parsed as "exclude `*rg.fish*`". I'll file a bug downstream.

**Comment by zanchey** [comment]
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=964385

**Comment by faho** [comment]
Note that whatever we do Ubuntu won't get the update - they still haven't upgraded to 3.1.*2*.

**Comment by zanchey** [comment]
Yes, I've noted that in https://bugs.launchpad.net/debian/+source/fish/+bug/1886659

**Comment by zanchey** [comment]
This is not a problem with the packages that we build, only in the Debian and derived packages.

**Comment by mithrandi** [comment]
Doh! Fixed in Debian 3.1.2-2
